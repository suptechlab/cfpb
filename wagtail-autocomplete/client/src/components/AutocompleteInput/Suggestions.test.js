import React from 'react';
import { shallow } from 'enzyme';

import Suggestions from './Suggestions';


const mockProps = {
  suggestions: [
    { id: 1, title: 'Alice' },
    { id: 2, title: 'Tekisha' },
  ],
  onChange: jest.fn(),
  onCreate: jest.fn(),
  onClick: jest.fn(),
  input: { value: '' },
  canCreate: true,
};


describe('Suggestions', () => {
  it('exists', () => {
    expect(Suggestions).toBeDefined();
  });

  it('mounts', () => {
    const suggestions = shallow(
      <Suggestions {...mockProps} />
    );
    expect(suggestions).toMatchSnapshot();
  });

  it('does not show create new if input value is blank', () => {
    const suggestions = shallow(
      <Suggestions
        {...mockProps}
        input={{ value: ' ' }}
        canCreate={true}
      />
    );

    const item = suggestions.findWhere(node => (
      node.type() === 'li' &&
      node.text().startsWith('Create new')
    ));
    expect(item.exists()).toEqual(false);
  });

  it('does show create new if input value is not blank', () => {
    const suggestions = shallow(
      <Suggestions
        {...mockProps}
        input={{ value: 'new item' }}
        canCreate={true}
      />
    );

    const item = suggestions.findWhere(node => (
      node.type() === 'li' &&
      node.text().startsWith('Create new') &&
      node.text().includes('new item')
    ));
    expect(item.exists()).toEqual(true);
  });

  it('sets the correct aria owns id', () => {
    const suggestions = shallow(
      <Suggestions {...mockProps} />
    );

    const ariaOwnsId = suggestions.find('input').prop('aria-owns');
    const id = suggestions.find('ul').prop('id');
    expect(id).toBeTruthy();
    expect(id).toEqual(ariaOwnsId);
  });

  it('sets the correct aria active descendant id', () => {
    const suggestions = shallow(
      <Suggestions {...mockProps} />
    );

    let activeDescendantId = suggestions.find('input').prop('aria-activedescendant');
    expect(activeDescendantId).toBeFalsy();

    suggestions.setState({ visible: true, index: 0 });
    activeDescendantId = suggestions.find('input').prop('aria-activedescendant');
    const selectedOptionId = suggestions.find('li').get(0).props.id;
    const differentOptionId = suggestions.find('li').get(1).props.id;
    expect(activeDescendantId).toBeTruthy();
    expect(differentOptionId).toBeTruthy();
    expect(activeDescendantId).toEqual(selectedOptionId);
    expect(selectedOptionId === differentOptionId).toEqual(false);
  });
});
