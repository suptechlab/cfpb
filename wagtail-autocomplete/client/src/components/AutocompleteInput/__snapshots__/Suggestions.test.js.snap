// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions mounts 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="c-wagtailautocomplete__suggestions__sub"
>
    <div
        className="c-wagtailautocomplete__search-input-container"
    >
        <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup="true"
            aria-owns="wagtailautocomplete-0"
            className="c-wagtailautocomplete__search"
            id={undefined}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="combobox"
            type="text"
            value=""
        />
        <SearchIcon
            className="c-wagtailautocomplete__search-icon"
        />
    </div>
    <ul
        className="c-wagtailautocomplete__suggestions"
        id="wagtailautocomplete-0"
        role="listbox"
        style={
            Object {
                "display": "none",
              }
        }
    >
        <li
            className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
            id="wagtailautocomplete-0-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="option"
        >
            <span>
                Alice
            </span>
            <RightArrowIcon />
        </li>
        <li
            className="c-wagtailautocomplete__suggestions__item"
            id="wagtailautocomplete-0-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="option"
        >
            <span>
                Tekisha
            </span>
            <RightArrowIcon />
        </li>
    </ul>
</span>,
  "nodes": Array [
    <span
      className="c-wagtailautocomplete__suggestions__sub"
>
      <div
            className="c-wagtailautocomplete__search-input-container"
      >
            <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-expanded={false}
                  aria-haspopup="true"
                  aria-owns="wagtailautocomplete-0"
                  className="c-wagtailautocomplete__search"
                  id={undefined}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="combobox"
                  type="text"
                  value=""
            />
            <SearchIcon
                  className="c-wagtailautocomplete__search-icon"
            />
      </div>
      <ul
            className="c-wagtailautocomplete__suggestions"
            id="wagtailautocomplete-0"
            role="listbox"
            style={
                  Object {
                        "display": "none",
                      }
            }
      >
            <li
                  className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                  id="wagtailautocomplete-0-0"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
            >
                  <span>
                        Alice
                  </span>
                  <RightArrowIcon />
            </li>
            <li
                  className="c-wagtailautocomplete__suggestions__item"
                  id="wagtailautocomplete-0-1"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
            >
                  <span>
                        Tekisha
                  </span>
                  <RightArrowIcon />
            </li>
      </ul>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Suggestions
        canCreate={true}
        input={
                Object {
                        "value": "",
                      }
        }
        onChange={[Function]}
        onClick={[Function]}
        onCreate={[Function]}
        suggestions={
                Array [
                        Object {
                          "id": 1,
                          "title": "Alice",
                        },
                        Object {
                          "id": 2,
                          "title": "Tekisha",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Suggestions {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleBlurClick": [Function],
        "handleFocus": [Function],
        "handleKeyPress": [Function],
        "props": Object {
          "canCreate": true,
          "input": Object {
            "value": "",
          },
          "onChange": [Function],
          "onClick": [Function],
          "onCreate": [Function],
          "suggestions": Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "index": 0,
          "visible": false,
        },
        "suggestionItemsElm": Object {},
        "suggestionsControlsId": "wagtailautocomplete-0",
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="c-wagtailautocomplete__suggestions__sub"
>
          <div
                    className="c-wagtailautocomplete__search-input-container"
          >
                    <input
                              aria-activedescendant=""
                              aria-autocomplete="list"
                              aria-expanded={false}
                              aria-haspopup="true"
                              aria-owns="wagtailautocomplete-0"
                              className="c-wagtailautocomplete__search"
                              id={undefined}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              role="combobox"
                              type="text"
                              value=""
                    />
                    <SearchIcon
                              className="c-wagtailautocomplete__search-icon"
                    />
          </div>
          <ul
                    className="c-wagtailautocomplete__suggestions"
                    id="wagtailautocomplete-0"
                    role="listbox"
                    style={
                              Object {
                                        "display": "none",
                                      }
                    }
          >
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                              id="wagtailautocomplete-0-0"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              role="option"
                    >
                              <span>
                                        Alice
                              </span>
                              <RightArrowIcon />
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item"
                              id="wagtailautocomplete-0-1"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              role="option"
                    >
                              <span>
                                        Tekisha
                              </span>
                              <RightArrowIcon />
                    </li>
          </ul>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="c-wagtailautocomplete__suggestions__sub"
>
          <div
                    className="c-wagtailautocomplete__search-input-container"
          >
                    <input
                              aria-activedescendant=""
                              aria-autocomplete="list"
                              aria-expanded={false}
                              aria-haspopup="true"
                              aria-owns="wagtailautocomplete-0"
                              className="c-wagtailautocomplete__search"
                              id={undefined}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              role="combobox"
                              type="text"
                              value=""
                    />
                    <SearchIcon
                              className="c-wagtailautocomplete__search-icon"
                    />
          </div>
          <ul
                    className="c-wagtailautocomplete__suggestions"
                    id="wagtailautocomplete-0"
                    role="listbox"
                    style={
                              Object {
                                        "display": "none",
                                      }
                    }
          >
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                              id="wagtailautocomplete-0-0"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              role="option"
                    >
                              <span>
                                        Alice
                              </span>
                              <RightArrowIcon />
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item"
                              id="wagtailautocomplete-0-1"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              role="option"
                    >
                              <span>
                                        Tekisha
                              </span>
                              <RightArrowIcon />
                    </li>
          </ul>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Suggestions
    canCreate={true}
    input={
        Object {
            "value": "",
          }
    }
    onChange={[Function]}
    onClick={[Function]}
    onCreate={[Function]}
    suggestions={
        Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ]
    }
/>,
}
`;
